apisix/plugin.lua

```diff
@@ -1273,22 +1329,26 @@ function _M.run_global_rules(api_ctx, global_rules, phase_name)
             api_ctx.global_rules = global_rules
         end
 
+        local dummy_global_rule = merge_global_rule_lrucache(conf_version, global_rules, merge_global_rules, global_rules)
+        if not dummy_global_rule then
+            core.log.error("failed to get merged global rules form cache, using merge_global_rules instead")
+            dummy_global_rule = merge_global_rules(global_rules)
```

Comment:
This is unnecessary, lruchche will automatically execute `merge_global_rules` when the request key does not exist.